### PR TITLE
[CST] Use json_schemer in SavedClaim

### PR DESCRIPTION
**Pull request by iandonovan**
## Summary

- This work is behind a feature toggle (flipper): **YES**
- We are looking at [resolving an issue with JSON Schema](https://dsva.slack.com/archives/C053U7BUT27/p1732119843278369). Our idea is to use the `json_schemer` gem instead, and we are using the `SavedClaim` model as its first touchpoint.
- I am on BMT-2.

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/98065
- See Slack conversation above for full context

## Testing done

- [X] *New code is covered by unit tests*

## Acceptance criteria

- [X]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [X]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [X]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: app/models/saved_claim.rb [hunk 1]
```diff
@@ -148,13 +150,43 @@ def va_notification?(email_template_id)
   private
 
   def validate_schema(schema)
```
Comment (mjknight50) [inline app/models/saved_claim.rb:152]: Can you please add some docblocks or comments to these functions?
e.g. 
/**
 * Adds two numbers together.
 *
 * @param int $a The first number.
 * @param int $b The second number.
 * @return int The sum of the two numbers.
 */

File: spec/models/form526_submission_spec.rb [hunk 2]
```diff
@@ -28,7 +28,7 @@
   let(:backup_submitted_claim_status) { nil }
 
   before do
-    Flipper.disable(:disability_compensation_production_tester)
```
Comment (mjknight50) [inline spec/models/form526_submission_spec.rb:31]: Did you mean to remove change the name of the flipper here?

File: app/models/saved_claim.rb [hunk 3]
```diff
@@ -148,13 +150,43 @@ def va_notification?(email_template_id)
...
+  end
+
+  def validate_schema_with_json_schemer(schema)
+    errors = JSONSchemer.validate_schema(schema).to_a
+    return [] if errors.empty?
+
+    reformatted_schemer_errors(errors)
+  end
+
+  def validate_schema_with_json_schema(schema)
```
Comment (mjknight50) [inline app/models/saved_claim.rb:175]: same as above, please add docblocks or comments

File: app/models/saved_claim.rb [hunk 4]
```diff
@@ -148,13 +150,43 @@ def va_notification?(email_template_id)
...
     JSON::Validator.fully_validate_schema(schema, { errors_as_objects: true })
   rescue => e
     Rails.logger.error('Error during schema validation!', { error: e.message, backtrace: e.backtrace, schema: })
     raise
   end
 
-  def validate_form(schema, clear_cache)
+  def validate_form_with_json_schemer(schema)
```
Comment (mjknight50) [inline app/models/saved_claim.rb:182]: same as above, please add docblocks or comments

File: app/models/saved_claim.rb [hunk 5]
```diff
@@ -164,6 +196,15 @@ def validate_form(schema, clear_cache)
     raise
   end
 
+  def reformatted_schemer_errors(errors)
```
Comment (mjknight50) [inline app/models/saved_claim.rb:199]: same as above, please add docblocks or comments

File: spec/models/form526_submission_spec.rb (hunk 2 above)
Comment (iandonovan) [reply spec/models/form526_submission_spec.rb:31]: Yes. The `:disability_compensation_production_tester` flipper does not need to be disabled in this section of the spec, and the `:validate_saved_claims_with_json_schemer` does.

Notably -- and this PR is still in draft, so it's something I will tackle before officially publishing it -- this method of disabling Flippers is now deprecated. [The flipper docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#FeatureTogglesGuide-Backendexample) stipulate that the new paradigm going forward is to use `allow(Flipper).to receive(:enabled?).with(:name_of_flag).and_return(true/false)`.

**Review by danlim715 — DISMISSED** [review]

**Review by mjknight50 — APPROVED** [review]
GTG

**Review by ericboehs — APPROVED** [review]

**Comment by iandonovan** [comment]
@mjknight50 @ericboehs Thank you for reviewing!

I see the Danger action here telling me to split this PR up. I don't think this PR is a good candidate for that; unless I'm missing something fundamental, this PR is a small code change and a lot of testing changes, so splitting it would result in a smaller PR that fails tests 😂 

I'm not seeing documentation about the process for getting an exception to that rule. Do you know how I can go about that?

**Pull request merged**